src/ServiceManagement/Compute/Commands.ServiceManagement/IaaS/PersistentVMs/RestartAzureVM.cs

```diff
@@ -48,13 +55,17 @@ public string Name
         }
 
         [Parameter(Mandatory = true,
-            ValueFromPipelineByPropertyName = true,
+            ValueFromPipeline = true,
             HelpMessage = "The Virtual Machine to restart.",
             ParameterSetName = RestartInputParameterSet)]
         [Parameter(Mandatory = true,
-            ValueFromPipelineByPropertyName = true,
+            ValueFromPipeline = true,
             HelpMessage = "The Virtual Machine to redeploy.",
             ParameterSetName = RedployInputParameterSet)]
+        [Parameter(Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The Virtual Machine to initiate maintenance.",
+            ParameterSetName = InitiateMaintenanceInputParameterSet)]
```

Comment:
@vanbasten2323 why VM parameter is not part of InitiateMaintenanceByNameParameterSet?